Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean

```diff
@@ -927,8 +928,73 @@ def localizationMapOfSpecializes {x y : PrimeSpectrum R} (h : x ⤳ y) :
         ⟨a, show a ∈ x.asIdeal.primeCompl from h ha⟩ : _))
 #align prime_spectrum.localization_map_of_specializes PrimeSpectrum.localizationMapOfSpecializes
 
+variable (R) in
+/--
+Zero loci of prime ideals are closed irreducible sets in the Zariski topology and any closed
+irreducible set is a zero locus of some prime ideal.
+-/
+protected def pointsEquivIrreducibleCloseds :
+    PrimeSpectrum R ≃o {s : Set (PrimeSpectrum R) | IsIrreducible s ∧ IsClosed s}ᵒᵈ where
+    __ := irreducibleSetEquivPoints (α := PrimeSpectrum R) |>.toEquiv |>.symm |>.trans
+      OrderDual.toDual
+    map_rel_iff' {p q} := show closure {q} ≤ closure {p} ↔ p ≤ q by
+      simp [closure_singleton, Set.le_eq_subset, zeroLocus_subset_zeroLocus_iff,
+        q.IsPrime.radical]
+
 end PrimeSpectrum
 
+open PrimeSpectrum in
+/--
+[Stacks: Lemma 00ES (3)](https://stacks.math.columbia.edu/tag/00ES)
+Zero loci of minimal prime ideals of `R` are irreducible components in `Spec R` and any
+irreducible component is a zero locus of some minimal prime ideal.
+-/
+protected def minimalPrimes.equivIrreducibleComponents :
+    minimalPrimes R ≃o (irreducibleComponents <| PrimeSpectrum R)ᵒᵈ :=
```

Comment:
I'll add some lemma/defs about OrderEmbedding/Iso and minimals/maximals to make this easy to construct using `OrderIso.setCongr`, `OrderIso.dual` together with [irreducibleComponents_eq_maximals_closed](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Irreducible.html#irreducibleComponents_eq_maximals_closed)

Reply:
With #9190 this becomes
```
  let e : {p : Ideal R | p.IsPrime ∧ ⊥ ≤ p} ≃o PrimeSpectrum R :=
    ⟨⟨fun x ↦ ⟨x.1, x.2.1⟩, fun x ↦ ⟨x.1, x.2, bot_le⟩, fun _ ↦ rfl, fun _ ↦ rfl⟩, Iff.rfl⟩
  (e.trans <| PrimeSpectrum.pointsEquivIrreducibleCloseds R).minimalsIsoMaximals.trans
    (OrderIso.setCongr _ _ <| by simp_rw [irreducibleComponents_eq_maximals_closed, and_comm]).dual
```

Reply:
> With #9190 this becomes
> 
> ```
>   let e : {p : Ideal R | p.IsPrime ∧ ⊥ ≤ p} ≃o PrimeSpectrum R :=
>     ⟨⟨fun x ↦ ⟨x.1, x.2.1⟩, fun x ↦ ⟨x.1, x.2, bot_le⟩, fun _ ↦ rfl, fun _ ↦ rfl⟩, Iff.rfl⟩
>   (e.trans <| PrimeSpectrum.pointsEquivIrreducibleCloseds R).minimalsIsoMaximals.trans
>     (OrderIso.setCongr _ _ <| by simp_rw [irreducibleComponent
> ```

Done, thanks for the suggestion!